### PR TITLE
Fix PABEE & PL CI failure

### PR DESCRIPTION
**Pull request by JetRunner**
#6421

### CODE REVIEW

**Review by LysandreJik — APPROVED** [review]
LGTM!

**Comment by LysandreJik** [comment]
Ah, PL test still failing!

**Comment by JetRunner** [comment]
@LysandreJik Don't worry. It seems I mistype some parameter name

**Comment by JetRunner** [comment]
Oops! @sshleifer could you have a look at the PL example? I've tried tweaking the parameters but it doesn't seem to work.

**Comment by JetRunner** [comment]
@stas00 Can you take a look? @sshleifer is on a vacation. Lots of thanks!

**Comment by stas00** [comment]
Yes, of course, I will be able to investigate in a few hours.

**Comment by stas00** [comment]
ok, so I had a look. I'm able to reproduce the problem of low `acc` with those changes proposed in this PR.

If you have a look at https://github.com/huggingface/transformers/pull/6034 I tried various params to no avail, it was working fine on my machine, but CI kept on failing. It was just very counterproductive trying to experiment w/o being able to reproduce it locally.

**Comment by JetRunner** [comment]
@stas00 Yes I've already found the problem in #6034 (output_dir) and fixed that in our PR. However the accuracy is still too low compared to the trainer version of run_glue. Since you can now reproduce the low acc, please give it a look!

**Comment by stas00** [comment]
Thank you for explaining what is happening, @JetRunner 

I have no perms to push, so try to use this:
```
        testargs = """
            run_pl_glue.py
            --model_name_or_path bert-base-cased
            --data_dir ./tests/fixtures/tests_samples/MRPC/
            --task mrpc
            --do_train
            --do_predict
            --output_dir ./tests/fixtures/tests_samples/pl_temp_dir
            --train_batch_size=32
            --learning_rate=1e-4
            --num_train_epochs=4
            --warmup_steps=3
            --seed=42
            --max_seq_length=128
            """.split()`
```
I get acc/f1 of 1.0 with this config, the key was more `--num_train_epochs` and some warm-up.

So you uncovered that these tests are very unreliable as they don't clean up after themselves and re-runs give invalid results. It's enough to get one config that succeeded, all the test re-runs will succeed at the moment. At the very least pl_glue needs to support `--overwrite_output_dir`.

**Comment by JetRunner** [comment]
@stas00 this perm also outputs `0.5`, sadly. I feel maybe there's another bug here in the PL example? 
cc @LysandreJik @sshleifer

**Comment by JetRunner** [comment]
PABEE's bug is fixed in #6453. The reproducible low acc is still existing for PL.
cc @LysandreJik @sshleifer
